junit5-support/src/main/kotlin/io/specmatic/test/SpecmaticJUnitSupport.kt

```diff
@@ -101,7 +115,7 @@ open class SpecmaticJUnitSupport {
         }
 
         private fun getReportConfiguration(): ReportConfiguration {
-            return when (val reportConfiguration = specmaticConfig?.report) {
+            return when (val reportConfiguration = getReport(specmaticConfig)) {
```

Comment:
getReport (which is in SpecmaticConfig.Companion) is meant to be used only within configuration classes. It looks like it's presence is being checked (first `null` branch of the `when`). You can use a method on SpecmaticConfig to answer that question instead of returning the object.